src/gauging.jl

```diff
@@ -0,0 +1,58 @@
+"""Modify the diagonal elements of an ITensor with a specified function"""
+function modify_diagonal_els_itensor(A::ITensor, f::Function)
+  out = copy(A)
+  is = [[j for i in 1:length(inds(A))] for j in 1:mindim(out)]
+  for i in is
+    out[i...] = f(out[i...])
+  end
+
+  return out
+end
+
+"""Put an ITensorNetwork into the symmetric gauge and also return the message tensors (which are the diagonal bond matrices)"""
+function symmetrise_itensornetwork(ψ::ITensorNetwork)
```

Comment:
What about just `symmetric_gauge(ψ::ITensorNetwork)`? `symmetrise` is quite a general term (i.e. it could mean forcing the tensors to be symmetric about permutations of the indices, which is not what this is doing in general, though for a network that is actually rotationally invariant maybe it helps).

Reply:
Yeah that sounds like a good name. Happy with that.